### DISCUSSION TITLE
Incremental Hyperparameter optimization in GPy classifier

### DISCUSSION THREAD

**Discussion by ishcha** [SheffieldML/GPy#939] [Q&A] [answered]
Hi!

Is there any way to do an epoch-wise incremental gradient descent hyperparameter optimization for the Gaussian Process class [GPy.core.gp](https://gpy.readthedocs.io/en/deploy/GPy.core.html#module-GPy.core.gp) under the GPy package? I am familiar with the complete optimization function model.optimize(), but unable to find any clue for incremental learning, as is supported by partial_fit() methods in sklearn estimators. 

Any clue or help in this is highly appreciated. 

Thanks in advance!

**Comment by zhenwendai** [comment, marked as answer]
GPy model memorizes the parameter values so that if you keep the same GPy model instance, ```model.optimize()``` will start from the previous parameters. If you do not want to optimize until convergence, you can specify the maximum number of iterations.

**Reply by ishcha** [reply]
Thank you very much for your reply. 
Can we possibly change the Y_metadata parameter in GPy.core.gp class, after the model declaration? I found methods to change X and Y, but I am unable to change Y_metadata, with the same shape as Y.
